### PR TITLE
Separate Declarations and Statements

### PR DESCRIPTION
**Pull request by ikarienator**

### CODE REVIEW

File: spec.idl [hunk 1]
```diff
@@ -470,6 +471,10 @@ interface CatchClause : Node {
   attribute Block body;
 };
 
+interface ConstDeclaration : Delcaration {
```
Comment (michaelficarra) [inline spec.idl]: Typo.

File: spec.idl [hunk 2]
```diff
@@ -478,6 +483,10 @@ interface Identifier : Node {
   attribute string name;
 };
 
+interface LetDeclaration : Delcaration {
```
Comment (michaelficarra) [inline spec.idl]: Typo.

File: spec.idl [hunk 3]
```diff
@@ -470,6 +471,10 @@ interface CatchClause : Node {
   attribute Block body;
 };
 
+interface ConstDeclaration : Delcaration {
+  attribute VariableDeclaration declaration;
```
Comment (michaelficarra) [inline spec.idl]: This should be a list of declarators. It should be a VariableDeclaration analogue.

**Comment by michaelficarra** [comment]
`Export`, `ForInStatement`, `ForOfStatement`, and `ForStatement` all need to support `LetDeclaration` and `ConstDeclaration`.

**Comment by ikarienator** [comment]
Ping @michaelficarra

**Comment by michaelficarra** [comment]
I don't like this any more. :-1:.

**Pull request closed without merging**